agent/host_agent_test.go

```diff
@@ -369,6 +388,69 @@ var _ = Describe("Agent", func() {
 		})
 	})
 
+	Context("When --version flag is created using 'version.sh' script", func() {
+		var (
+			tmpHostAgentBinary string
+			gitMajor           string
+			gitMinor           string
+			gitVersion         string
+			err                error
+		)
+		BeforeEach(func() {
+			command := exec.Command("/bin/sh", "-c", "git describe --tags --abbrev=14 --match 'v[0-9]*' 2>/dev/null")
+			command.Stderr = os.Stderr
+			cmdOut, _ := command.Output()
+			gitVersion = strings.TrimSuffix(string(cmdOut), "\n")
+
+			gitVersion = strings.Split(gitVersion, "-")[0]
+			gitVars := strings.Split(gitVersion, ".")
+			if len(gitVars) > 1 {
+				gitMajor = gitVars[0][1:]
+				gitMinor = gitVars[1]
+			}
+
+			root, _ := exec.Command("/bin/sh", "-c", "git rev-parse --show-toplevel").Output()
+			cmd := exec.Command("/bin/sh", "-c", strings.TrimSuffix(string(root), "\n")+"/hack/version.sh")
+			ldflags, _ := cmd.Output()
+			tmpHostAgentBinary, err = gexec.Build("github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/agent", "-ldflags", string(ldflags))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			tmpHostAgentBinary = ""
+			gitMajor = ""
+			gitMinor = ""
+			gitVersion = ""
+		})
+
+		It("should match local generated git values", func() {
+			expectedStruct := version.Info{
+				Major:      gitMajor,
+				Minor:      gitMinor,
+				GitVersion: gitVersion,
+			}
+			expected := fmt.Sprintf("byoh-hostagent version: %#v\n", expectedStruct)
+			out, err := exec.Command(tmpHostAgentBinary, "--version").Output()
+			Expect(err).NotTo(HaveOccurred())
+
+			re := regexp.MustCompile(`Major:\"(.*?)\"`)
+			majorOut := re.Find(out)
+			majorExpected := re.Find([]byte(expected))
+
+			re = regexp.MustCompile(`Minor:\"(.*?)\"`)
+			minorOut := re.Find(out)
+			minorExpected := re.Find([]byte(expected))
+
+			re = regexp.MustCompile(`GitVersion:\"(.*?)\"`)
+			gitVersionOut := re.FindSubmatch(out)
+			gitVersionExpected := re.FindSubmatch([]byte(expected))
+
+			Expect(majorOut).Should(Equal(majorExpected))
+			Expect(minorOut).Should(Equal(minorExpected))
```

Comment:
meta - the assertion should be closer to where its variables are defined / assigned / evaluated. That is, move the corresponding assertions closer to majorOut / minorOut blocks

Reply:
Yeah looks much better that way. Will do.